### PR TITLE
added 429 check, fix #144

### LINKED ISSUE #144: preventing 429 errors
If this is triggered, we should "sleep" fo a long enough period. ratelimit is 300 calls in 5 minutes so we can simply sleep 5 minutes and should be good

### PR DESCRIPTION
**Pull request by schochastics**

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -33,73 +28,87 @@ print.rtoot_bearer <- function(x,...){
...
+    if (is.null(instance)) {
+        token <- check_token_rtoot(token)
+        url <- prepare_url(token$instance)
+        config <- httr::add_headers(Authorization = paste("Bearer", token$bearer))
+    } else {
+        url <- prepare_url(instance)
+        config <- list()
+    }
+    count <- 0
+    while (TRUE) {
```
Comment (schochastics) [inline R/utils.R:55]: this is where the change happened.

**Comment by schochastics** [comment]
sorry for the indent mess. I will move to 4 indent for the entire repo after merging this

File: R/utils.R [hunk 2]
```diff
@@ -33,73 +28,87 @@ print.rtoot_bearer <- function(x,...){
...
+        if (!status_code %in% c(200, 429)) {
+            stop(paste("something went wrong. Status code:", status_code), call. = FALSE)
+        } else if (status_code == 429) {
+            message("too many requests. Sleeping for 5 minutes")
+            Sys.sleep(60 * 5)
+            count <- count + 1
+        } else if (status_code == 200) {
+            break()
+        }
+    }
```
Comment (schochastics) [inline R/utils.R:74]: change stops here

**Review by chainsawriot — APPROVED** [review]
LGTM.

**Comment by chainsawriot** [comment]
Copy this to here, maybe.

https://github.com/gesistsa/sweater/blob/v0.1/.editorconfig

**Pull request merged**